### ISSUE TITLE
TCC linking failed (darwin)

### ISSUE THREAD

**Issue by ttdennis** [frida/frida#1688]
It seems that tcc has issues when linking a CModule (at least on iOS/macOS). Using tcc, the following results in an error. This is an extremely dumbed down example, but I think it suffices.

```
[Local::test]-> a=new CModule("void *mmap(void*, int, int, int, int, int); int main() {mmap(0, 0x10, 0, 0, 0, 0);}");
Error: linking failed: tcc: error: undefined symbol '_mmap'
```

When using clang it seems to work:
```
[Local::test]-> a=new CModule("void *mmap(void*, int, int, int, int, int); int main() {mmap(0, 0x10, 0, 0, 0, 0);}", {}, { toolchain: '
external' });
{
    "main": "0x104215f70"
}
```

Additionally, when using frida-core in version `14.0.8` this problem doesn't come up, tcc properly links the module (see [here](https://github.com/ttdennis/fpicker/blob/main/harness/darwin-shm.js)).

I couldn't really find out where exactly the issue is, but I think the general problem is that the symbol name should be `mmap` instead of `_mmap`.

**Comment by oleavr** [comment]
Ah yes this is indeed quite confusing.

CModule only provides a very limited number of built-in symbols, basically only the ones declared [here](https://github.com/frida/frida-gum/tree/master/bindings/gumjs/runtime/cmodule-tcc) and [here](https://github.com/frida/frida-gum/tree/master/bindings/gumjs/runtime/cmodule). Previously TinyCC would call `dlsym(RTLD_DEFAULT, name)` whenever it encountered an undefined symbol, but this behavior changed when we moved to the current version. I found the previous behavior quite confusing (and not portable to Windows), so I considered it a bug and not a feature. :) So basically you will have to pass in the address of `mmap` yourself through the constructor's second argument.

However when using the external toolchain, the whole libc is available. This is not really consistent, but it's probably not worth preventing that.
